projects/rocprofiler-sdk/source/include/rocprofiler-sdk/agent.h

```diff
@@ -117,6 +117,37 @@ ROCPROFILER_CXX_CODE(
         sizeof(rocprofiler_agent_runtime_visiblity_t) == sizeof(uint32_t),
         "Increasing the size of the rocprofiler_agent_runtime_visiblity_t is not permitted");)
 
+/**
+ * @brief Firmware version information for a GPU agent.
+ *
+ * This struct contains the firmware version numbers for various GPU microcontrollers and engines.
+ * All fields are 32-bit unsigned integers representing the firmware version as reported by the
+ * corresponding sysfs files in /sys/class/drm/renderD{}/device/fw_version/
+ */
+typedef struct rocprofiler_agent_fw_info_v0_t
+{
+    uint64_t size;                 ///< size of this struct
+    uint32_t mec2_fw_version;      ///< Firmware version for MEC2 engine
+    uint32_t mec_fw_version;       ///< Firmware version for MEC engine
+    uint32_t rlc_fw_version;       ///< Firmware version for RLC engine
+    uint32_t rlc_srlc_fw_version;  ///< Firmware version for RLC SRLC engine
+    uint32_t rlc_srlg_fw_version;  ///< Firmware version for RLC SRLG engine
+    uint32_t rlc_srls_fw_version;  ///< Firmware version for RLC SRLS engine
+    uint32_t sdma2_fw_version;     ///< Firmware version for SDMA2 engine
+    uint32_t sdma_fw_version;      ///< Firmware version for SDMA engine
+    uint32_t smc_fw_version;       ///< Firmware version for SMC engine
+    uint32_t sos_fw_version;       ///< Firmware version for Secure OS (PSP)
+    uint32_t ta_ras_fw_version;    ///< Firmware version for TA RAS microcontroller
+    uint32_t ta_xgmi_fw_version;   ///< Firmware version for TA XGMI microcontroller
+    uint32_t vcn_fw_version;       ///< Firmware version for VCN engine
+} rocprofiler_agent_fw_info_v0_t;
+
+/**
+ * @brief Typedef for the current ::rocprofiler_agent_fw_info_v0_t
+ *
+ */
+typedef rocprofiler_agent_fw_info_v0_t rocprofiler_agent_fw_info_t;
```

Comment:
```suggestion
typedef struct rocprofiler_agent_firmware_info_v0_t
{
    uint32_t mec2_version;      ///< Firmware version for MEC2 engine
    uint32_t mec_version;       ///< Firmware version for MEC engine
    uint32_t rlc_version;       ///< Firmware version for RLC engine
    uint32_t rlc_srlc_version;  ///< Firmware version for RLC SRLC engine
    uint32_t rlc_srlg_version;  ///< Firmware version for RLC SRLG engine
    uint32_t rlc_srls_version;  ///< Firmware version for RLC SRLS engine
    uint32_t sdma2_version;     ///< Firmware version for SDMA2 engine
    uint32_t sdma_version;      ///< Firmware version for SDMA engine
    uint32_t smc_version;       ///< Firmware version for SMC engine
    uint32_t sos_version;       ///< Firmware version for Secure OS (PSP)
    uint32_t ta_ras_version;    ///< Firmware version for TA RAS microcontroller
    uint32_t ta_xgmi_version;   ///< Firmware version for TA XGMI microcontroller
    uint32_t vcn_version;       ///< Firmware version for VCN engine
} rocprofiler_agent_firmware_info_v0_t;
```
No need for `size` since this will not be expanded in future. Get rid of typedef since this type will always be directly used.

Reply:
You should document that a value of `0` means no version info was found. Also, recommend changing `*_fw_version` to `*_version` since the struct name makes it clear all these versions are related to the firmware... it is just redundant

Reply:
sure, addressed all the above and previous suggestions. using firmware version as 0xFFFFFFFF as the invalid firmware